src/main/app/claims/responseModelConverter.ts

```diff
@@ -10,86 +13,205 @@ import { Company } from 'app/claims/models/details/yours/company'
+  }
+}
+
+function convertCommon (draft: ResponseDraft) {
+  return {
+    defendant: convertPartyDetails(draft.defendantDetails),
+    freeMediation: draft.freeMediation && draft.freeMediation.option,
+    moreTimeNeeded: draft.moreTimeNeeded && draft.moreTimeNeeded.option,
+    statementOfTruth: mapOptional(convertStatementOfTruth)(draft.qualifiedStatementOfTruth)
+  }
+}
+
+function convertStatementOfTruth (draft: QualifiedStatementOfTruth): StatementOfTruth {
+  return new StatementOfTruth(
+    draft.signerName,
+    draft.signerRole
+  )
+}
+
+function convertPartyDetails (defendant: Defendant): Party {
+  let party: Party = undefined
+  switch (defendant.partyDetails.type) {
+    case PartyType.INDIVIDUAL.value:
+      party = new Individual()
+      if ((defendant.partyDetails as IndividualDetails).dateOfBirth) {
+        (party as Individual).dateOfBirth = (defendant.partyDetails as IndividualDetails).dateOfBirth.date.asString()
+      }
+      break
+    case PartyType.COMPANY.value:
+      party = new Company()
+      if ((defendant.partyDetails as CompanyDetails).contactPerson) {
+        (party as Company).contactPerson = (defendant.partyDetails as CompanyDetails).contactPerson
+      }
+      break
+    case PartyType.ORGANISATION.value:
+      party = new Organisation()
+      if ((defendant.partyDetails as OrganisationDetails).contactPerson) {
+        (party as Organisation).contactPerson = (defendant.partyDetails as OrganisationDetails).contactPerson
+      }
+      break
+    case PartyType.SOLE_TRADER_OR_SELF_EMPLOYED.value:
+      party = new SoleTrader()
+      if ((defendant.partyDetails as SoleTraderDetails).businessName) {
+        (party as SoleTrader).businessName = (defendant.partyDetails as SoleTraderDetails).businessName
+      }
+      break
+  }
+  party.address = new Address().deserialize(defendant.partyDetails.address)
+  if (defendant.partyDetails.hasCorrespondenceAddress) {
+    party.correspondenceAddress = new Address().deserialize(defendant.partyDetails.correspondenceAddress)
+  }
+  party.name = defendant.partyDetails.name
+  if (defendant.email) {
+    party.email = defendant.email.address
+  }
+  if (defendant.mobilePhone) {
+    party.mobilePhone = defendant.mobilePhone.number
+  }
+  return party
+}
+
+function convertDefenceType (draft: RejectAllOfClaim): DefenceType {
+  switch (draft.option) {
+    case RejectAllOfClaimOption.ALREADY_PAID:
+      return DefenceType.ALREADY_PAID
+    case RejectAllOfClaimOption.DISPUTE:
+      return DefenceType.DISPUTE
+  }
+
+  throw new Error(`Unsupported defence type: ${draft.option}`)
+}
+
+function convertPartAdmissionType (draft: RejectPartOfClaim): PartAdmissionType {
+  switch (draft.option) {
+    case RejectPartOfClaimOption.AMOUNT_TOO_HIGH:
+      return PartAdmissionType.AMOUNT_TOO_HIGH
+    case RejectPartOfClaimOption.PAID_WHAT_BELIEVED_WAS_OWED:
+      return PartAdmissionType.PAID_WHAT_BELIEVED_WAS_OWED
+  }
```

Comment:
Shouldn't we throw an error when conversion failed? In theory it looks like very unlikely scenario because task list should not allow submitting response without making sure draft is valid. It is more to fail fast before making call to backend which would return 400.